tests/trustless_gateway_car_test.go

```diff
@@ -374,10 +374,60 @@ func TestTrustlessCarDagScopeAll(t *testing.T) {
 }
 
 func TestTrustlessCarEntityBytes(t *testing.T) {
-	singleLayerHamtMultiBlockFilesFixture := car.MustOpenUnixfsCar("t0118/single-layer-hamt-with-multi-block-files.car")
-	subdirWithMixedBlockFiles := car.MustOpenUnixfsCar("t0118/subdir-with-mixed-block-files.car")
+	singleLayerHamtMultiBlockFilesFixture := car.MustOpenUnixfsCar("trustless_gateway_car/single-layer-hamt-with-multi-block-files.car")
+	subdirWithMixedBlockFiles := car.MustOpenUnixfsCar("trustless_gateway_car/subdir-with-mixed-block-files.car")
+	missingBlockFixture := car.MustOpenUnixfsCar("trustless_gateway_car/file-3k-and-3-blocks-missing-block.car")
 
 	tests := SugarTests{
+		{
+			Name: "GET CAR with entity-bytes succeeds even if the gateway is missing a block after the requested range",
+			Hint: `
+				dag-scope=entity&entity-bytes=0:x should return a CAR file with
+				only the blocks needed to fullfill the request. This MUST
+				succeed despite the fact that bytes beyond 'x' are not retrievable.
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", missingBlockFixture.MustGetCidWithCodec(0x70)).
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "0:1000"),
+			Response: Expect().
+				Status(200).
+				Body(
+					IsCar().
+						IgnoreRoots().
+						HasBlocks(
+							missingBlockFixture.MustGetCid(),
+							missingBlockFixture.MustGetChildrenCids()[0],
+						).
+						Exactly(),
+				),
+		},
+		{
+			Name: "GET CAR with entity-bytes succeeds even if the gateway is missing a block before the requested range",
+			Hint: `
+				dag-scope=entity&entity-bytes=y:* should return a CAR file with
+				only the blocks needed to fullfill the request. This MUST
+				succeed despite the fact that a block with bytes before 'y' is
+				not retrievable.
+			`,
+			Request: Request().
+				Path("/ipfs/{{cid}}", missingBlockFixture.MustGetCidWithCodec(0x70)).
+				Query("format", "car").
+				Query("dag-scope", "entity").
+				Query("entity-bytes", "2200:*"),
```

Comment:
:information_source:  I had gut feeling to check what we send on the wire – we may need to add `QueryRaw` here, but not in scope of this PR, not a blocker –  filled separate https://github.com/ipfs/gateway-conformance/issues/88